packages/events/TopLevelEventTypes.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {DOMTopLevelEventType} from 'react-dom/src/events/DOMTopLevelEventTypes';
```

Comment:
By importing the type from the `react-dom` module, we can use an opaque type and avoid leaking the underlying type in any other module.

Defining the opaque type in `DOMTopLevelEventTypes` makes the types a lot easier but introduces this not-so-nice type dependency on `react-dom`.